dbt/adapters/synapse/connections.py

```diff
@@ -16,3 +16,22 @@ class SynapseConnectionManager(SQLServerConnectionManager):
     TYPE = "synapse"
     TOKEN = None
 
+    @classmethod
+    def get_response(cls, cursor) -> AdapterResponse:
+        #message = str(cursor.statusmessage)
+        message = 'OK'
+        rows = cursor.rowcount
+        #status_message_parts = message.split() if message is not None else []
+        #status_messsage_strings = [
+        #    part
+        #    for part in status_message_parts
+        #    if not part.isdigit()
+        #]
+        #code = ' '.join(status_messsage_strings)
+        return AdapterResponse(
+            _message=message,
+            #code=code,
+            rows_affected=rows
+        )
```

Comment:
@jtcohen6 looks like `get_response()` is an abstract base method, so it must be implemented?  At least that was the error I was getting in the new conda env where I installed the package fresh and ran it... Perhaps my installation was wrong? Or is there a way to inherit from `SQLServerCredentials`? I don't see a `get_response` method implemented for `dbt-redshift` which makes me thing I'm doing something wrong...
 
 I'm a big Python OOP noob, maybe something like this?

```suggestion
    @classmethod
    def get_response(cls, cursor) -> AdapterResponse:
        return cls.get_response
    )
```

Reply:
You're right, dbt-redshift inherits the implementation of [`get_response` from dbt-postgres](https://github.com/fishtown-analytics/dbt/blob/42a85ac39f34b058678fd0c03ff8e8d2835d2808/plugins/postgres/dbt/adapters/postgres/connections.py#L150-L165).

I'd _think_ that, by inheriting from `SQLServerConnectionManager` in `SynapseConnectionManager`, you'd be able to avoid reimplementing `get_response` (unless you wanted it to do something genuinely different here).

Reply:
yeah we don't to do anything different. must be the same issue mikael was having...